common/source/kernel/Syscall.cpp

```diff
@@ -123,7 +126,7 @@ size_t Syscall::close(size_t fd)
 
 size_t Syscall::open(size_t path, size_t flags)
 {
-  if (path >= USER_BREAK)
+  if (path >= USER_BREAK || strlen((const char*) path) > MAX_PATH_LENGTH)
```

Comment:
do we want to keep this not thread safe?

Reply:
The problem I see here is that it demonstrates an unsafe example for students who might just copy the check without thinking about it